lib/eosknowledgeprivate/ekn-media-bin.c

```diff
@@ -1401,13 +1408,64 @@ ekn_media_bin_update_position (EknMediaBin *self)
   gtk_label_set_label (priv->audio_position_label, format_time (position));
 }
 
+static inline void
+log_fps (EknMediaBin *self)
+{
+  EknMediaBinPrivate *priv = EMB_PRIVATE (self);
+  GdkFrameClock *clock;
+  GstSample *sample;
+  GstBuffer *buffer;
+  gdouble time;
+
+  g_object_get (priv->play, "sample", &sample, NULL);
+
+  if (sample)
+    {
+      buffer = gst_sample_get_buffer (sample);
+      gst_sample_unref (sample);
+    }
+
+  if (priv->last_buffer == buffer)
+    return;
+
+  clock = gtk_widget_get_frame_clock (GTK_WIDGET (self));
+  priv->last_buffer = buffer;
+  priv->frames_window++;
+
+  time = (gdk_frame_clock_get_frame_time (clock) - priv->tick_start) / 1000000.0;
+
+  if (priv->playback_time == (gint)time)
```

Comment:
You're doing a lot of double/int conversions, and you're also using `guint`.

You should just use `gint64` everywhere and keep the value in milliseconds.